dsl/pace/dsl/stencil_config.py

```diff
@@ -216,36 +217,6 @@ def __eq__(self, other):
         except AttributeError:
             return False
 
-    def _get_backend_opts(
-        self,
-        device_sync: Optional[bool] = None,
-        format_source: Optional[bool] = None,
-    ) -> Dict[str, Any]:
-        backend_opts: Dict[str, Any] = {}
-        all_backend_opts: Optional[Dict[str, Any]] = {
-            "device_sync": {
-                "backend": r".*(gpu|cuda)$",
-                "value": False,
-            },
-            "format_source": {
-                "value": False,
-            },
-            "verbose": {"backend": r"(gt:|cuda)", "value": False},
-        }
-        for name, option in all_backend_opts.items():
-            using_option_backend = re.match(
-                option.get("backend", ""), self.compilation_config.backend
-            )
-            if "backend" not in option or using_option_backend:
-                backend_opts[name] = option["value"]
-
-        if device_sync is not None:
-            backend_opts["device_sync"] = device_sync
-        if format_source is not None:
-            backend_opts["format_source"] = format_source
-
-        return backend_opts
-
```

Comment:
This was a really confusing function, and we didn't end up ever using the confusing logic here, since we always passed `device_sync` and `format_source` anyway. I replaced this with a simple constant dict that this would have returned.